base/errorshow.jl

```diff
@@ -359,8 +359,10 @@ function show_method_candidates(io::IO, ex::MethodError, @nospecialize kwargs=()
         end
     end
 
-    for (func, arg_types_param) in funcs
-        for method in methods(func)
+    for (f, arg_types_param) in funcs
+        f2 = f isa UnionAll ? unwrap_unionall(unwrap_unionall(f).name.wrapper) : f
```

Comment:
`unwrap_unionall` doesn't necessarily have a field `name` (if it's not a DataType). Most nearly, you can get `typename(f)`, although that also may throw errors

Reply:
I will check for `unwrap_unionall(f) isa DataType`.

Only to be curious: which cases are there, where `f` is a `UnionAll` and `unwrap_unionall(f)` is not a `DataType`?